src/code/z_play.c

```diff
@@ -15,6 +15,14 @@ u64 D_801614D0[0xA00];
 
 void Play_SpawnScene(PlayState* this, s32 sceneNum, s32 spawn);
 
+// If R_ENABLE_PLAY_LOGS is enabled, the number "1" with a file and line number will be printed during
+// various phases of the Play update cycle. This can be helpful for debugging at what stage a problem may be occuring.
```

Comment:
It feels a bit unnecessary to mention the "Play update cycle" here. I think I would just say something like "will be printed at various points", or simply when the macro is used.
It's also slightly incorrect since the log is used in `Play_Main` as well

Reply:
`Play_Main` is the "state update" function I guess, but yeah it's more common to think of update as `Play_Update`

I also wouldn't mention how this can be helpful for debugging, as this applies to nearly every single print/debug feature imo

Reply:
I feel like it's worth mentioning, mainly because on the surface it may not be so clear why the number "1" is being spammed everywhere. It's not exactly a self explanatory logging system imo. 
If each print was descriptive and said what was about to happen, it'd be different story

I think saying something like "the play update cycle" is important cause of how "top level" all of this is. A log call between two functions is really a log call across entire systems being processed, so I really want to stress that importance. I know 3/43 are in main, but it's still its main purpose. But if there's another way to say that and still convey its importance, lmk

Reply:
To me this explanation on how it helps debugging Play_Main, feels the same as if you had a comment in an actor like "if you want to change how the actor behaves you can do this".

Maybe it's just the wording that feels like it's directly addressed to the reader that puts me off?
Wording what you remarked on on discord, that the logs happen at a high level, between each sub-system "update", sounds more useful/detailed to me

Reply:
even in my original comment I dont see how Im addressing the reader. Its phrased in an impersonal way without acknowledging the reader at all, but maybe I am misunderstanding 

But yeah, I am trying to convey the point that the logs happen in between play sub systems being processed. My current wording is: `// This can be helpful for debugging at what point during the `PlayState` loop an issue is occurring.`
That is expecting that you have that pretext that the play state is a gamestate and is processing many systems. But Id rather make that assumption than try and be specific about the places that it gets used. Which is why I say "it can be used to.."

Reply:
Maybe a suggestion will clear up what I mean?
Current:
```c
// If R_ENABLE_PLAY_LOGS is enabled, the number "1" with a file and line number will be printed.
// This can be helpful for debugging at what point during the `PlayState` loop an issue is occuring.
```
->
```c
// If R_ENABLE_PLAY_LOGS is enabled, the number "1" with a file and line number will be printed at various key points in
// the play state execution, in particular between calls to various systems.
// As a result, this allows tracing what the play state execution is doing at a high level.
```

(explain what it does and why it's neat instead of just mentioning it's good for debugging (most debug features are useful, given you are in a situation that could use them))

Idk maybe another way to express how I feel about giving such tips in the source is that it feels out of place, comments should document and explain but not give an opinion or tips (for example we don't have "set this to E for 20fps" in the rom header when that's really one of the first things anyone should do)

Reply:
I dont see my current wording as giving tips or advice. I see it has explaining, objectively, why it is helpful in this context. I only use the phrasing of *can be* because it depends on how its used.

For example, your comment is assuming that they *are* placed inbetween calls to systems. Which they are, but Id rather say that they can be helpful if used in that way, rather than comment on how the original devs used it directly. Its explaining its purpose for existing, and if used right, will be helpful.

I do not agree that im saying "debug feature is good", I am saying that this particular debug feature has a helpful purpose if used right.

All that said, I will try to take your comment and work it in a way that works for me

Reply:
```
// If R_ENABLE_PLAY_LOGS is enabled, the number "1" with a file and line number will be printed.
// When placed at various key points in the Play state execution, like between calls to various systems, 
// it can allow for tracing what the play state execution is doing at a high level.
```
for example, is how I would phrase this.
My point is I want to say what the macros *purpose* is, while not saying "this macro *is used* here. That kind of phrasing, to me, is like commenting on where certain functions get used. The point of the macro description here is to explain how it can be used, not how it is used.

Hopefully this wording is a suitable compromise. I think we just have different perspectives on what kind of angle these descriptions should be coming from.

Reply:
Sorry to argue over such a small thing, I really don't understand your point of view :sweat: , I hope it doesn't look like bad faith

> When placed at various key points

This feels weird because it makes it sound like it's up to a dev to add the line wherever, but it's already there?

Also I understand we don't want to explicitly document where something is used, but for regs/debug features in particular it seems appropriate to indicate what they seem to be for
For example just the reg being named after "play" indicates being about the play state / z_play ?

Reply:
we'll just need more opinions to tiebreaker this, cause I dont think we'll reach an agreement

Reply:
I think I agree with Fig about trying to say what the macro's purpose is and not where it's used in the base code. I think I would even go as far as only stating what the macro does and the general purpose of such a log (ie. tracing back to a specific function call or section of code), without referring to play itself, but I understand if that's seen as too vague.

Regarding Fig's suggestion and the part about `When placed at various key points`, I agree it can sound a bit weird, but maybe that can be improved with a different wording like `By being placed at ...`? Tho I don't think it's a big deal either way, personally